source/validate_id.cpp

```diff
@@ -264,6 +264,30 @@ bool idUsage::isValid<SpvOpTypeSampler>(const spv_instruction_t*,
   return true;
 }
 
+// True if the integer constant is > 0. constWords are words of the
+// constant-defining instruction (either OpConstant or
+// OpSpecConstant). typeWords are the words of the constant's-type-defining
+// OpTypeInt.
+bool aboveZero(const std::vector<uint32_t>& constWords,
+               const std::vector<uint32_t>& typeWords) {
+  const auto width = typeWords[2];
+  const bool is_signed = typeWords[3];
+  if (width == 64) {
+    if (is_signed) {
+      int64_t value = constWords[3] | (int64_t{constWords[4]} << 32);
```

Comment:
Thanks @yavn.  I changed it to `uint64_t`, as you suggest.  Aside from the VS2013 error, I've realized that shifting a signed value runs into undefined behaviour all too easily.